plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -896,14 +903,14 @@ int process_and_send(struct flb_cloudwatch *ctx, const char *input_plugin,
              * Iterate through the record map, extract intermediate metric data, 
              * and add to the list.
              */
-            for (i = 0; i < map_size; i++) {
```

Comment:
The variable `i` is used to track the number of logs included in payload. `j` should be used to avoid overwriting `i`
's value via loop iterator.

Reply:
good catch